src/tests/util.rs

```diff
@@ -139,8 +139,15 @@ pub trait RequestHelper {
     /// Issue a PUT request
     #[track_caller]
     fn put<T>(&self, path: &str, body: impl Into<Bytes>) -> Response<T> {
+        let body = body.into();
+        let is_json = body.starts_with(b"{") && body.ends_with(b"}");
```

Comment:
I feel like this will eventually cause someone to have an annoying day writing tests if we ever add an endpoint that doesn't expect an object, but presumably that's not an issue right now.

I guess the other option would be to add a parallel `put_json` method that sets the header. I don't feel very strongly about this. :shrug:

Reply:
yeah, though arguably APIs shouldn't return raw arrays anyway and that extra friction might be good 😄

the only PUT that does not deal with Json currently is the publish endpoint. it felt not worth it to have a dedicated method just for that.